authentication/src/main/java/io/camunda/authentication/CamundaOAuthPrincipalService.java

```diff
@@ -116,14 +117,20 @@ public OAuthContext loadOAuthContext(final Map<String, Object> claims)
     if (StringUtils.hasText(groupsClaim)) {
       groups = new HashSet<>(oidcGroupsLoader.load(claims));
     } else {
-      // TODO: Get groups for username and clientId https://github.com/camunda/camunda/issues/26572
       groups =
-          groupServices.getGroupsByMemberIds(mappingIds, MAPPING).stream()
+          groupServices
+              .findAll(
+                  GroupQuery.of(
+                      groupQuery ->
+                          groupQuery.filter(
+                              groupFilter -> groupFilter.memberIdsByType(ownerTypeToIds))))
```

Comment:
🔧 I think it'd be easier to understand this code if we put it in a method in the group services. Same for roles tbh 😄 

Imo overloading the existing method (if it's even used still) is fine: `groupServices.getGroupsByMemberIds(ownerTypeToIds)`